onnxruntime/core/session/environment.cc

```diff
@@ -55,16 +55,51 @@ Status Environment::Create(std::unique_ptr<logging::LoggingManager> logging_mana
   return status;
 }
 
+namespace OrtMemoryInfoUtils {
+
+// Ugly but necessary for instances where we want to check equality of two OrtMemoryInfos
+// without accounting for OrtAllocatorType in the equality checking process.
+// TODO: Should we remove the OrtAllocatorType field from the OrtMemoryInfo struct to
+// avoid such problems and also remove the unintuitive phenomenon of binding an allocator
+// type to OrtMemoryInfo (which loosely is just device info) ?
+bool Equals(
```

Comment:
I tired to do it but it turned out to be pretty complicated. I don't know if I am missing something very obvious.  `Equals()` calls the `== operator` which is defined after the struct in the header. This `== operator`s  function definition can't be moved before the `OrtMemoryInfo` struct either because it needs the struct's definition..